src/interface/metadata.hpp

```diff
@@ -54,10 +48,44 @@
   PARTHENON_INTERNAL_FOR_FLAG(Edge)                                                      \
   /**  node variable */                                                                  \
   PARTHENON_INTERNAL_FOR_FLAG(Node)                                                      \
-  /**  a vector quantity, i.e. a rank 1 contravariant tenso */                           \
+  /************************************************/                                     \
+  /**  ROLE: Exactly one must be specified (default is Provides) */                      \
+  /**  Private to a package */                                                           \
+  PARTHENON_INTERNAL_FOR_FLAG(Private)                                                   \
+  /**  Provided by a package */                                                          \
+  PARTHENON_INTERNAL_FOR_FLAG(Provides)                                                  \
+  /**  Not created by a package, assumes available from another package */               \
+  PARTHENON_INTERNAL_FOR_FLAG(Requires)                                                  \
+  /**  does nothing if another package provides the variable */                          \
+  PARTHENON_INTERNAL_FOR_FLAG(Overridable)                                               \
+  /************************************************/                                     \
+  /**  SHAPE: Neither or one (but not both) can be specified */                          \
+  /**  a vector quantity, i.e. a rank 1 contravariant tensor */                          \
   PARTHENON_INTERNAL_FOR_FLAG(Vector)                                                    \
   /**  a rank-2 tensor */                                                                \
   PARTHENON_INTERNAL_FOR_FLAG(Tensor)                                                    \
+  /************************************************/                                     \
+  /**  DATATYPE: Exactly one must be specified (default is Real) */                      \
+  /** Boolean-valued quantity */                                                         \
```

Comment:
Not all data types are supported for all variables. Bool and integer variables are only used in swarms.

Reply:
You're right, it doesn't seem to be used. I'm happy to remove it unless there are plans to use it in the future

Reply:
It looks like `Metadata` doesn't know whether it's used for swarms or fields, so I think I'll leave this part. But I'll add a check in `CellVariable` to make sure the `Metadata` says `Real`.